src/quo2/components/navigation/top_nav/view.cljs

```diff
@@ -0,0 +1,134 @@
+(ns quo2.components.navigation.top-nav.view
+  (:require [react-native.core :as rn]
+            [quo2.components.buttons.button.view :as button]
+            [quo2.components.avatars.user-avatar.view :as user-avatar]
+            [quo2.components.counter.counter.view :as counter]
+            [quo2.components.navigation.top-nav.style :as style]
+            [quo2.theme :as quo.theme]
+            [quo2.components.common.notification-dot.view :as notification-dot]))
+
+(defn- get-button-common-props
+  [{:keys [jump-to? theme blur?]}]
+  {:icon-only?      true
+   :size            32
+   :container-style {:margin-left 12}
+   :type            (cond
+                      jump-to?                          :black
+                      (and (not blur?) (= :dark theme)) :dark-grey
+                      :else                             :grey)
+   :background      (when blur? :blur)})
+
+(defn- unread-indicator
+  [{:keys [notification-count max-value customization-color]}]
+  (when (pos? notification-count)
+    [counter/view
+     {:accessibility-label :activity-center-unread-count
+      :type                :default
+      :customization-color customization-color
+      :container-style     (style/unread-indicator notification-count
+                                                   max-value)}
+     notification-count]))
+
+(defn unread-dot
+  [{:keys [customization-color blur? notification]}]
+  [notification-dot/view
+   {:style               style/unread-dot
+    :blur?               blur?
+    :customization-color (when (= notification :notification) customization-color)}])
+
+(defn notification-highlight
+  [{:keys [notification notification-count
+           max-unread-notifications customization-color]
+    :as   props}]
+  (case notification
```

Comment:
Also this `case` can be

(when (#{:seen :notification :mention} notification})
[unread-dot props])